uniffi/src/bindings/kotlin/templates/ErrorTemplate.kt

```diff
@@ -0,0 +1,96 @@
+@Structure.FieldOrder("code", "message")
```

Comment:
All this is copied from [`RustError`](https://github.com/mozilla/application-services/blob/main/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/rust/RustError.kt)

Not sure if there are big design decisions that maybe should change?

Seemed okay to me, only big difference is that I'm defining an `Error` and an `Exception` for each error type. An alternative is to have only one `Error` and use the codes to map to different `Exception`s. 

Thoughts?

Reply:
So since we have support for multiple errors enums, each with variants on the Rust side, I'm matching each error there to an error here (and swift), and then using the error codes to match variants to exceptions.

Another alternative I'm very happy to change (it won't be much of a change because of the macros ❤️ ) Is to only support one error on the Rust/IDL side, that would put a limitation there, but keep us to one error here (similar to how it's done in A-S)